src/niscope/metadata/functions_addon.py

```diff
@@ -0,0 +1,85 @@
+    'FetchArrayMeasurement':           { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'FetchBinary16':                   { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'FetchBinary32':                   { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'FetchBinary8':                    { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'FetchComplex':                    { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'FetchComplexBinary16':            { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+    'Read':                            { 'codegen_method': 'no',       },  # Has niScope_wfmInfo param
+}
+
+# Attach the given parameter to the given enum from enums.py
+functions_enums = {
+    # @TODO add all enums
+}
+
+# This is the additional information needed by the code generator to properly generate the buffer retrieval mechanism
+# {'is_buffer': True} is required for all parameters that are arrays. Some were able to be detected as an array when
+#   generating functions.py. This sets 'is_buffer' for those parameters where the dectection didn't work
+# {'size': <size information>} is required for all output buffers.
+# <size information> is a dictionary with two keys: 'mechanism' and 'value'.
+#   'mechanism' can be:
+#       'fixed':        The size is known ahead of time, usually defined by the API.
+#                       'value' should be an int.
+#       'passed-in':    When the size comes from another parameter.
+#                       'value' should be the name of the parameter through which this is specified.
+#       'ivi-dance':    When the size is determined by calling into the function using a size of zero and
+#                       interpreting the return value as a size rather than an error.
+#                       'value' should be the name of the parameter through which the size (0, then the real
+#                       one) is passed in. This parameter won't exist in the corresponding Python Session method.
+functions_buffer_info = {
+    'GetError':                     { 'parameters': { 3: { 'size': {'mechanism':'ivi-dance', 'value':'bufferSize'}, }, }, },
+    'self_test':                    { 'parameters': { 2: { 'size': {'mechanism':'fixed', 'value':256}, }, }, }, # From documentation
+    'GetAttributeViString':         { 'parameters': { 4: { 'size': {'mechanism':'ivi-dance', 'value':'bufSize'}, }, }, },
+    'GetCalUserDefinedInfo':        { 'parameters': { 1: { 'size': {'mechanism':'fixed', 'value':256}, }, }, }, # From LabVIEW VI, even though niDMM_GetCalUserDefinedInfoMaxSize() exists.
+    'error_message':                { 'parameters': { 2: { 'size': {'mechanism':'fixed', 'value':256}, }, }, }, # From documentation
+    'init':                         { 'parameters': { 0: { 'is_buffer': True, }, }, },
```

Comment:
'is_buffer': True should not be necessary anymore.

Reply:
Removed